src/Components/Operations.js

```diff
@@ -0,0 +1,164 @@
+import { Component } from "react";
+import './Operations.css'
+import React from 'react'
+import CloseIcon from '@material-ui/icons/Close'
+import moment from 'moment'
+
+class Operations extends Component {
+
+    constructor() {
+        super()
+        this.state = {
+            amountInput: '',
+            vendorInput: '',
+            categoryInput: '',
+            dateInput: moment().format('YYYY-MM-DD'),
+            showSuccess: false,
+            showNotEnough: false,
+            showProblem: false
+        }
+    }
+
+    handleInputChange = event => {
+        const value = event.target.value;
+        const name = event.target.name;
+
+        this.setState({
+            [name]: value
+        });
+    }
+
+    isValidDate(d) {
+        return d instanceof Date && !isNaN(d);
+    }
+    
+    deposit = () => {
+        if (!this.state.amountInput ||
+            !this.state.vendorInput ||
+            !this.state.categoryInput) {
+            this.setState({
+                showProblem: true
+            })
+        } else {
+            this.props.deposit({
+                amount: this.state.amountInput,
+                vendor: this.state.vendorInput,
+                category: this.state.categoryInput,
+                date: this.state.dateInput
+            })
+            this.setState({
+                amountInput: '',
+                vendorInput: '',
+                categoryInput: '',
+                dateInput: moment().format('YYYY-MM-DD'),
+                showSuccess: true
+            })
+        }
+    }
+
+    withdraw = () => {
+        if (!this.state.amountInput ||
+            !this.state.vendorInput ||
+            !this.state.categoryInput) {
+            this.setState({
+                showProblem: true
+            })
+        } else if (this.props.balance - this.state.amountInput < 500) {
+            this.setState({
+                showNotEnough: true
+            })
+        } else {
+            this.props.withdraw({
+                amount: this.state.amountInput,
+                vendor: this.state.vendorInput,
+                category: this.state.categoryInput,
+                date: this.state.dateInput
+            })
+            this.setState({
+                amountInput: '',
+                vendorInput: '',
+                categoryInput: '',
+                dateInput: moment().format('YYYY-MM-DD'),
+                showSuccess: true
+            })
+        }
+    }
+
+    handleSnackbarClose = () => {
+        this.setState({
+            showNotEnough: false,
+            showSuccess: false,
+            showProblem: false
+        })
+    }
+
+    render() {
+        return (
+            <div id="operations">
+                <Box component="div" display="inline">
+                    <TextField size="small" type="number" value={this.state.amountInput} name="amountInput" onChange={this.handleInputChange} label="Amount" />
```

Comment:
You could always make an input component and render it using a map for each type of input